pkg/dashboards/kubernetes/apiserver/apiserver.go

```diff
@@ -84,42 +84,53 @@ func withAPIServerResources(datasource string, clusterLabelMatcher *labels.Match
 	)
 }
 
-func BuildAPIServerOverview(project string, datasource string, clusterLabelName string) dashboards.DashboardResult {
-	clusterLabelMatcher := dashboards.GetClusterLabelMatcher(clusterLabelName)
-	clusterLabelMatcherV2 := dashboards.GetClusterLabelMatcherV2(clusterLabelName)
-	return dashboards.NewDashboardResult(
-		dashboard.New("api-server-overview",
-			dashboard.ProjectName(project),
-			dashboard.Name("Kubernetes / API server"),
-			dashboard.AddVariable("cluster",
-				listVar.List(
-					labelValuesVar.PrometheusLabelValues("cluster",
-						labelValuesVar.Matchers("up{"+panels.GetAPIServerMatcher()+"}"),
-						dashboards.AddVariableDatasource(datasource),
-					),
-					listVar.DisplayName("cluster"),
+func BuildAPIServerOverview(project string, datasource string, clusterLabelName string, variableOverrides []dashboard.Option) dashboards.DashboardResult {
```

Comment:
How about making the variableOverrides a variadic option, so that you don't always have to pass in an empty slice if you aren't using custom variables?
main.go highlights the problem. 🙂 

```suggestion
func BuildAPIServerOverview(project string, datasource string, clusterLabelName string, variableOverrides ...dashboard.Option) dashboards.DashboardResult {
```